src/main/java/cn/nukkit/entity/item/EntityMinecartAbstract.java

```diff
@@ -175,8 +175,10 @@ public boolean onUpdate(int currentTick) {
             if (Rail.isRailBlock(block)) {
                 processMovement(dx, dy, dz, (BlockRail) block);
                 if (block instanceof BlockRailActivator) {
-                    // Activate the minecart/TNT
-                    activate(dx, dy, dz, (block.getDamage() & 0x8) != 0);
+                    if (((BlockRailActivator) block).isActive()){
```

Comment:
This would be more appropriate.
`if (block instanceof BlockRailActivator && ((BlockRailActivator) block).isActive()){`